wiring/src/spark_wiring_ble.cpp

```diff
@@ -577,81 +576,81 @@ BleAdvertisingData::BleAdvertisingData(const iBeacon& beacon)
 
 size_t BleAdvertisingData::set(const uint8_t* buf, size_t len) {
     if (buf == nullptr || len == 0) {
-        selfLen_ = 0;
-        return selfLen_;
+        selfData_.clear();
+        return selfData_.size();
     }
-    len = std::min(len, (size_t)BLE_MAX_ADV_DATA_LEN);
-    memcpy(selfData_, buf, len);
-    selfLen_ = len;
-    return selfLen_;
+    selfData_.clear();
+    len = std::min(len, (size_t)BLE_MAX_ADV_DATA_LEN_EXT);
+    selfData_.append(buf, len);
+    return selfData_.size();
 }
 
 size_t BleAdvertisingData::set(const iBeacon& beacon) {
     clear();
 
     if (beacon.UUID().type() == BleUuidType::SHORT) {
-        return selfLen_;
+        return selfData_.size();
     }
 
-    selfData_[selfLen_++] = 0x02;
-    selfData_[selfLen_++] = BLE_SIG_AD_TYPE_FLAGS;
-    selfData_[selfLen_++] = BLE_SIG_ADV_FLAGS_LE_ONLY_GENERAL_DISC_MODE;
-    selfData_[selfLen_++] = 0x1a; // length
-    selfData_[selfLen_++] = BLE_SIG_AD_TYPE_MANUFACTURER_SPECIFIC_DATA;
+    CHECK_TRUE(selfData_.reserve(BLE_MAX_ADV_DATA_LEN), 0);
```

Comment:
We should use `BLE_MAX_ADV_DATA_LEN_EXT_CONNECTABLE` here.

Reply:
By specification, iBeacon data is 30 bytes in length.